projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/kinit_authenticator.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+import tempfile
+from subprocess import call
+
+from vdk.internal.core import errors
+from vdk.plugin.kerberos.base_authenticator import BaseAuthenticator
+
+log = logging.getLogger(__name__)
+
+
+class KinitGSSAPIAuthenticator(BaseAuthenticator):
+    def __init__(
+        self, keytab_pathname: str, kerberos_principal: str, working_dir: str = None
+    ):
+        super().__init__(keytab_pathname, kerberos_principal)
+
+        self._working_dir = working_dir
+        self.__configure_current_os_process_to_use_own_kerberos_credentials_cache()
+
+    def __repr__(self):
```

Comment:
is to string used anywhere? seeing a lot of investment with representing the object as string, yet I am not sure about the use case

Reply:
I don't think it is used but it could still be useful when debugging.